### PR TITLE
React Implementation - Sprint 2

### PR DESCRIPTION
**Pull request by OliOliKT**

### CODE REVIEW

File: src/components/AccountInformation.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      <h2>Account Information</h2>
+      <div className="info-item">
+        <p>
+          <strong>User ID</strong>
+        </p>
+        <p>
+          {showUserId ? (
+            <span>{currentUser.id}</span>
+          ) : (
+            <span>*************</span>
```
Comment (mircealungu) [inline src/components/AccountInformation.js:31]: why are we showing stars when the uid is not defined?

File: src/components/AddCommentButton.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import "./AddCommentButton.css"; // The corresponding CSS file
+
+function AddCommentButton() {
+  return (
+    <div className="add-comment-button">
+        <p className="add-comment-text"> + Add comment </p>
```
Comment (mircealungu) [inline src/components/AddCommentButton.js]: a component should be reusable. 
This is just a hardcoded button with the text that's not changeable.
why is i a component then?

File: src/components/ButtonAndText.css [hunk 3]
```diff
@@ -1,25 +1,22 @@
...
+.iconAndText:hover .fa-mug-hot,
+.iconAndText .fa-mug-hot.liked  {
+  color: hsl(152, 93%, 65%);
+}
+
+.iconAndText:hover .iconText {
+  color: hsl(152, 93%, 65%);
+}
+
+.iconLink {
```
Comment (mircealungu) [inline src/components/ButtonAndText.css:21]: kebab-case?

File: src/components/CommentComponent.js [hunk 4]
```diff
@@ -1,49 +1,68 @@
-import React from "react";
+import React, { useState } from "react";
 import "./CommentComponent.css";
 
-const WriteComment = () => {
+const WriteComment = ({ postId, closeCommentModal, onCommentPosted }) => {
```
Comment (mircealungu) [inline src/components/CommentComponent.js:6]: You never name a component FooComponent. It's clearly a component based on it's structure.

File: src/components/CommentButtonAndText.js [hunk 5]
```diff
@@ -1,13 +1,15 @@
...
 import React from "react";
 import "./ButtonAndText.css";
+import { Link } from "react-router-dom"; 
 
-function CommentButtonAndText() {
+function CommentButtonAndText({ commentCount, onComment }) {
   return (
-    <div className="iconAndText">
-      <a href="#" className="iconLink">
+    <div className="iconAndText" onClick={onComment}>
```
Comment (mircealungu) [inline src/components/CommentButtonAndText.js:8]: name all event handlers with `handle` maybe? as you do in the next file?

File: src/components/CommentComponent.js [hunk 6]
```diff
@@ -1,49 +1,68 @@
...
+
+  const handleCommentSubmit = async () => {
+    // replace this logic with database callbacks.
+    console.log("Comment submitted:", comment, "for postId:", postId);
+    onCommentPosted();
+    closeCommentModal();
+  };
+
   return (
     <section className="writeComment">
```
Comment (mircealungu) [inline src/components/CommentComponent.js:44]: based on the code we see here there's no need for this component to exist. simply move the section inside of the CommentBox. OR/?

File: src/components/CreateUserButtonAndText.js [hunk 7]
```diff
@@ -0,0 +1,73 @@
+import { useState } from "react";
```
Comment (mircealungu) [inline src/components/CreateUserButtonAndText.js]: looks good.

File: src/components/FormInput.js [hunk 8]
```diff
@@ -0,0 +1,66 @@
+import React from "react";
+import "./FormInput.css";
+import { useState } from "react";
+import Parse from "parse";
+import { useNavigate } from "react-router-dom";
+import { Link } from "react-router-dom";
+
+function LogInButtonAndFormInput() {
```
Comment (mircealungu) [inline src/components/FormInput.js:8]: it's good practice to have same name for component and container file. (https://github.com/itu-tid/lecture-notes/blob/main/Tech-TopUps/5-Coding-Practices.md#component-and-file-in-which-it-is-defined-should-have-the-same-name)

File: src/components/FormInput.js [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+
+  const handleLogin = async () => {
+    setError("");
+    try {
+      const user = await Parse.User.logIn(email, password);
+      console.log("Success! User ID:", user.id);
+      navigate("/DiscoverFeed");
+    } catch (error) {
+      console.error("Error while logging in user", error);
+      setError("You typed in the wrong password!");
```
Comment (mircealungu) [inline src/components/FormInput.js]: your error message might be misleading at times. do you know why?

File: src/components/Post.js [hunk 10]
```diff
@@ -1,46 +1,69 @@
...
 import './Post.css';
+import WriteComment from './CommentComponent';
+import { useNavigate } from 'react-router-dom'; // Import useHistory
+
+function Post({ postTitle, mood, postedBy, postContent, postId, sipCount, currentUser }) {
+  const [isSipped, setIsSipped] = useState(false);
+  const [updatedSipCount, setUpdatedSipCount] = useState(sipCount); // State to store updated sipCount
+  const navigate = useNavigate();
+
+
```
Comment (mircealungu) [inline src/components/Post.js]: these many empty lines tell me that you're not using format-on-save with Prettier (https://github.com/itu-tid/lecture-notes/blob/main/Tech-TopUps/5-Coding-Practices.md#install-an-automatic-formatter-and-configure-it-to-format-on-save)

File: src/components/Post.js [hunk 11]
```diff
@@ -1,46 +1,69 @@
...
+  const handleCommentIconClick = () => {
+    console.log(postId); // test to see if we log the id and not "undefined"
+    navigate(`/posts/${postId}`, { state: { postTitle, mood, postedBy, postContent, postId } });
+  };
+
+  const handleSip = async () => {
+    try {
+      const Post = Parse.Object.extend("Post");
+      const query = new Parse.Query(Post);
```
Comment (mircealungu) [inline src/components/Post.js:21]: i'm not sure i understand the 'sip' behavior. based on what i see here it alternates between 0 and 1?

File: src/components/TwoFactorAuthentication.js [hunk 12]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import './TwoFactorAuthentication.css';
+
+const TwoFactorAuthentication = () => {
```
Comment (mircealungu) [inline src/components/TwoFactorAuthentication.js:5]: :) i don't think you'll have time to implement that...

File: src/components/writePost.js [hunk 13]
```diff
@@ -0,0 +1,142 @@
...
+const WritePost = () => {
+  const [title, setTitle] = useState("");
+  const [post, setPost] = useState("");
+  const [userId, setUserId] = useState("unknown user");
+  const [mood, setMood] = useState("");
+
+  useEffect(() => {
+    async function fetchUserData() {
+      try {
+        const currentUser = Parse.User.current();
```
Comment (mircealungu) [inline src/components/writePost.js:15]: this is a very roundabout way of getting the username.
why do you have to get the session? 
you have the user id in the `Parser.User.current()`. You can simply query the User table based on that. 
or am I missing something?

**Review by mircealungu — CHANGES_REQUESTED** [review]
overall very good work!
please have a look at the comments and address the ones you can and agree with. 

mircea and @igawaclawska

File: src/components/AccountInformation.js (hunk 1 above)
Comment (jdisitu) [reply src/components/AccountInformation.js:31]: We have a hide button which toggles between the user id and the stars.

File: src/components/AddCommentButton.js (hunk 2 above)
Comment (jdisitu) [reply src/components/AddCommentButton.js]: resolved

File: src/components/ButtonAndText.css (hunk 3 above)
Comment (jdisitu) [reply src/components/ButtonAndText.css:21]: on it

File: src/components/CommentComponent.js (hunk 4 above)
Comment (jdisitu) [reply src/components/CommentComponent.js:6]: on it

File: src/components/CommentButtonAndText.js (hunk 5 above)
Comment (jdisitu) [reply src/components/CommentButtonAndText.js:8]: on it

File: src/components/FormInput.js (hunk 8 above)
Comment (jdisitu) [reply src/components/FormInput.js:8]: on it

File: src/components/FormInput.js (hunk 9 above)
Comment (jdisitu) [reply src/components/FormInput.js]: on it

File: src/components/Post.js (hunk 10 above)
Comment (jdisitu) [reply src/components/Post.js]: on it

**Pull request merged**